### DISCUSSION TITLE
Should jose catch base64 decode errors?

### DISCUSSION THREAD

**Discussion by maximivanov** [panva/jose#134] [Q&A] [answered]
Thanks for the library Filip!

Not sure if it's intended to be this way so I thought I better ask.

In a browser environment, I've tried to call `jose/jwt/verify` with malformed JWT (basically valid JWT minus 1 character).  As a result I've got "DOMException: Failed to execute 'atob' on 'Window': The string to be decoded is not correctly encoded.".

Is this expected or should it rethrow something more user-friendly e.g. https://github.com/panva/jose/blob/master/docs/classes/_util_errors_.jwtinvalid.md?

**Comment by panva** [comment, marked as answer]
@maximivanov I think it may re-throw as a TypeError instead. I'm hesitant to assign a JOSE specific error or code. To be released with the next fix/feature release. https://github.com/panva/jose/commit/c886f77c6a80150f36462a133c486855149217b6
